url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/dataStructure/PositionTable.java

```diff
@@ -0,0 +1,72 @@
+package dataStructure;
+
+import java.io.*;
+import java.util.*;
+
+public class PositionTable {
+    int size;
+    PositionTableEntry[] ptable;
+
+    public PositionTable (int tablesize) {
+	size = tablesize;
+
+	ptable = new PositionTableEntry[size];
+	for (int i =0; i < size; i++)
+	    ptable[i] = new PositionTableEntry();
+    }
+
+    public void insertEntry (int i, double x, double y) {
+	ptable[i].setX(x);
+	ptable[i].setY(y);
+    }
+
+    public double getX (int i) { return ptable[i].getX(); }
+
+    public double getY (int i) { return ptable[i].getY(); }
+
+    public void readFromTopologyFile (String fname) {
+	String s;
+
+	try {
+	    BufferedReader in = new BufferedReader(new FileReader(fname));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 31, 35, 44, 54, 55, 56. The resource is closed at line 58. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).